charmcraft/commands/store/__init__.py

```diff
@@ -769,15 +769,23 @@ def fill_parser(self, parser):
     def run(self, parsed_args):
         """Run the command."""
         store = Store(self.config.charmhub)
+        packages_list = store.list_registered_names(include_collaborations=True)
+        packages_type_map = {p.name: p.entity_type for p in packages_list}
```

Comment:
Since we're iterating over packages here we could just cherry-pick the one we want instead of building the full map. (Is there a way to query a specific package instead of downloading the full list?)

Reply:
I don't see any API can do that (https://api.charmhub.io/docs/default.html), but we can only keep the one that matched name.

Reply:
Updated.